include/zim/writer/creator.h

```diff
@@ -51,8 +51,13 @@ namespace zim
      * The current (documented) conditions when a exception is thrown are:
      * - When a entry cannot be added (mainly because a entry with the same path has already been added)
      *    A `zim::InvalidEntry` will be thrown. The creator will still be in a valid state and the creation can continue.
-     * - A incoherence has been detected in user implementation (of Item, ContentProvider, IndexData, ...)
-     *    A `zim::IncoherentImplementationError` will be thrown.
+     * - An exception has been thrown in a worker thread.
+     *    This exception will be catch and rethrown through a `zim::AsyncError`.
+     *    The creator will be set in a invalid state and creation cannot continue.
+     * - The creator is in error state.
+     *    A `zim::CreatorStateError` will be thrown.
```

Comment:
These two lines better belong to the commit where `CreatorStateError` is introduced